treeherder/seta/update_job_priority.py

```diff
@@ -0,0 +1,202 @@
+    for job in runnable_jobs_data['results']:
+        if not valid_platform(job['platform']):
+            continue
+
+        testtype = parse_testtype(
+            build_system_type=job['build_system_type'],
+            job_type_name=job['job_type_name'],
+            platform_option=job['platform_option'],
+            ref_data_name=job['ref_data_name']
+        )
+
+        if not testtype:
+            continue
+
+        # NOTE: This is *all* the data we need from the runnable API
+        new_job = {
+            'build_system_type': job['build_system_type'],  # e.g. {buildbot,taskcluster,*}
+            'platform': job['platform'],  # e.g. windows8-64
+            'platform_option': job['platform_option'],  # e.g. {opt,debug}
+            'testtype': testtype,  # e.g. web-platform-tests-1
+        }
+        key = _unique_key(new_job)
+
+        # Let's build a map of all the jobs and if duplicated change the build_system_type to *
+        if key not in job_build_system_type:
+            job_build_system_type[key] = job['build_system_type']
+            sanitized_list.append(new_job)
+        elif new_job['build_system_type'] != job_build_system_type[key]:
+            new_job['build_system_type'] = job_build_system_type[key]
+            # This will *replace* the previous build system type with '*'
+            # This guarantees that we don't have duplicates
+            sanitized_list[sanitized_list.index(new_job)]['build_system_type'] = '*'
+
+    return sanitized_list
+
+def query_sanitized_data(repo_name='mozilla-inbound'):
+    """Return sanitized jobs data based on runnable api. None if failed to obtain or no new data.
+
+     We need to find the latest gecko decision task ID (by querying the index [1][2])
+     in order to know which task ID to pass to the runnable api [3].
+
+     It stores the minimal sanitized data from runnable apis under ~/.mozilla/seta/<task_id>.json
+
+     [1] https://index.taskcluster.net/v1/task/gecko.v2.%s.latest.firefox.decision/
+     [2] Index's data structure:
+      {
+        "namespace": "gecko.v2.mozilla-inbound.latest.firefox.decision",
+        "taskId": "Dh9ZvFk5QCSprJ877cgUmw",
+        "rank": 0,
+        "data": {},
+        "expires": "2017-10-06T18:30:18.428Z"
+      }
+     [3] https://treeherder.mozilla.org/api/project/mozilla-inbound/runnable_jobs/?decision_task_id=Pp7ZxoH0SKyU6wnhX_Fp0g&format=json  # flake8: noqa
+    """
+    runnable_jobs = RunnableJobsClient().query_runnable_jobs(repo_name)
+    return _sanitize_data(runnable_jobs)
+
+def _two_weeks_from_now():
+    return datetime.datetime.now() + datetime.timedelta(days=14)
+
+def _initialize_values():
+    LOG.info('Fetch all rows from the job priority table.')
+    # Get all rows of job priorities
+    jp_index = job_priority_index(JobPriority.objects.all())
+    if jp_index:
+        # For 2 weeks the job will be considered high value (priority=1)
+        return jp_index, 1, 0, _two_weeks_from_now()
```

Comment:
Instead of hardcoding this, you should make this configurable via the treeherder settings (treeherder/config/settings.py), perhaps as SETA_HIGH_VALUE_INTERVAL or something.

Advantages to this:

* Easier to configure from heroku in case we need to tweak it
* Easier to override in a unit test if you need to (it seems like you do)